ui/common/src/socket.ts

```diff
@@ -350,14 +352,66 @@ export default class StrongSocket {
 
   pingInterval = (): number => this.computePingDelay() + this.averageLag;
   getVersion = (): number | false => this.version;
+
```

Comment:
these three methods are the bulk of the socket test code. we have three phases.

1. Track lag stats in memory with `updateStats`

2. On orderly connection close (rare) or 'pagehide' (most of our terminations), we call `storeStats` which dumps the current in-memory stats to local storage. it also calls `flushStats` but this is only relevant for the orderly close case because that method is async and task & microtask scheduling are no longer available in a pagehide event. 

3. On page loads, we call `flushStats` which checks local storage for stats from the previous session and sends them to idb (which is async). if the test has ended and there are values in local storage or idb, they are posted in one big batch to the collection endpoint and cleaned up on success.